[CI/CD][flaky test] fileconsumer test hits timeout on actuated ARM runner

### Component(s)

pkg/stanza

### Describe the issue you're reporting

[CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694389961/job/23843241727?pr=32175)

As with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32298, it looks like runs on actuated arm builds simply take a little longer sometimes, hitting timeout errors. It may be worthwhile to investigate ways to make the test suite finish quicker. Note: The specific test that this is failing on is likely irrelevant to the actual issue, it was simply the test being executed when the timeout occurred.

Failure output:
```
✖  fileconsumer (5m0.055s)

=== Failed
ERROR rerun aborted because previous run had a suspected panic and some test may not have run
=== FAIL: fileconsumer  (0.00s)
panic: test timed out after 5m0s
running tests:
	TestCopyTruncate (4m53s)

goroutine 1136 [running]:
testing.(*M).startAlarm.func1()
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:2366 +0x1dc
created by time.goFunc
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/time/sleep.go:177 +0x44

goroutine 1 [chan receive, 4 minutes]:
testing.(*T).Run(0xc0001eb1e0, {0xc476b2, 0x10}, 0xc8f5b0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1750 +0x604
testing.runTests.func1(0xc0001eb1e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:2161 +0x84
testing.tRunner(0xc0001eb1e0, 0xc00031f9e8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
testing.runTests(0xc0001a4978, {0x1400140, 0x32, 0x32}, {0xc00031fa78?, 0x5cce4?, 0x1423e20?})
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:2159 +0x6e4
testing.(*M).Run(0xc0002795e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:2027 +0xb78
go.uber.org/goleak.VerifyTestMain({0xda6400, 0xc0002795e0}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/go.uber.org/goleak@v1.3.0/testmain.go:53 +0x48
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMain(...)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/package_test.go:13
main.main()
	_testmain.go:[149](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694389961/job/23843241727?pr=32175#step:8:150) +0x2b4

goroutine 58 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004ce1a0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestBuild(0xc0004ce1a0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/config_test.go:436 +0x34
testing.tRunner(0xc0004ce1a0, 0xc8f598)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 59 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004ce340)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestBuildWithSplitFunc(0xc0004ce340)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/config_test.go:644 +0x30
testing.tRunner(0xc0004ce340, 0xc8f5a8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 47 [semacquire, 4 minutes]:
sync.runtime_Semacquire(0xc0000d20d0?)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/runtime/sema.go:62 +0x2c
sync.(*WaitGroup).Wait(0xc0000d20c8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/sync/waitgroup.go:116 +0x78
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).Stop(0xc0000d20c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:71 +0x9c
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestCopyTruncate.func4()
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/rotation_test.go:54 +0x38
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestCopyTruncate(0xc0004c4d00)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/rotation_test.go:85 +0x728
testing.tRunner(0xc0004c4d00, 0xc8f5b0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 64 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004ceb60)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestDefaultBehaviors(0xc0004ceb60)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:37 +0x34
testing.tRunner(0xc0004ceb60, 0xc8f5c8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 67 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf040)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestNopEncodingDifferentLogSizes(0xc0004cf040)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:223 +0x2a4
testing.tRunner(0xc0004cf040, 0xc8f698)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 65 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004ced00)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestReadExistingLogs(0xc0004ced00)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:76 +0x34
testing.tRunner(0xc0004ced00, 0xc8f6a8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 66 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004ceea0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestReadUsingNopEncoding(0xc0004ceea0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:139 +0x2bc
testing.tRunner(0xc0004ceea0, 0xc8f6c0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 68 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf1e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestReadNewLogs(0xc0004cf1e0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:253 +0x34
testing.tRunner(0xc0004cf1e0, 0xc8f6b8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 69 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf380)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestReadExistingAndNewLogs(0xc0004cf380)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:279 +0x34
testing.tRunner(0xc0004cf380, 0xc8f6a0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 70 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf520)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestStartAtEnd(0xc0004cf520)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:303 +0x34
testing.tRunner(0xc0004cf520, 0xc8f6f0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 71 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf6c0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestStartAtEndNewFile(0xc0004cf6c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:330 +0x34
testing.tRunner(0xc0004cf6c0, 0xc8f6f8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 72 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cf860)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestNoNewline(0xc0004cf860)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:350 +0x34
testing.tRunner(0xc0004cf860, 0xc8f690)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 73 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cfa00)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestEmptyLine(0xc0004cfa00)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:372 +0x34
testing.tRunner(0xc0004cfa00, 0xc8f5e8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 74 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cfba0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultipleEmpty(0xc0004cfba0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:395 +0x34
testing.tRunner(0xc0004cfba0, 0xc8f678)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 75 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004cfd40)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestLeadingEmpty(0xc0004cfd40)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:422 +0x34
testing.tRunner(0xc0004cfd40, 0xc8f628)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 76 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c4000)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestSplitWrite(0xc0004c4000)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:446 +0x34
testing.tRunner(0xc0004c4000, 0xc8f6e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 77 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c41a0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestIgnoreEmptyFiles(0xc0004c41a0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:466 +0x34
testing.tRunner(0xc0004c41a0, 0xc8f620)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 78 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c4340)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestDecodeBufferIsResized(0xc0004c4340)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:493 +0x34
testing.tRunner(0xc0004c4340, 0xc8f5c0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 79 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c44e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultiFileSimple(0xc0004c44e0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:513 +0x34
testing.tRunner(0xc0004c44e0, 0xc8f660)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 80 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c4680)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultiFileSort(0xc0004c4680)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:535 +0x34
make[2]: *** [../../Makefile.Common:126: test] Error 3
make[1]: *** [Makefile:165: pkg/stanza] Error 2
make: *** [Makefile:113: gotest] Error 2
testing.tRunner(0xc0004c4680, 0xc8f668)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 81 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c4820)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultiFileSortTimestamp(0xc0004c4820)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:568 +0x34
testing.tRunner(0xc0004c4820, 0xc8f670)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 82 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c49c0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultiFileParallel_PreloadedFiles(0xc0004c49c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:602 +0x34
testing.tRunner(0xc0004c49c0, 0xc8f658)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 83 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c4b60)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMultiFileParallel_LiveFiles(0xc0004c4b60)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:643 +0x34
testing.tRunner(0xc0004c4b60, 0xc8f650)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 123 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c5520)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestManyLogsDelivered(0xc0004c5520)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:742 +0x34
testing.tRunner(0xc0004c5520, 0xc8f630)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 126 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c5a00)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestEncodings(0xc0004c5a00)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:884 +0x34
testing.tRunner(0xc0004c5a00, 0xc8f5f0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 124 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c56c0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestFileBatching(0xc0004c56c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:776 +0x34
testing.tRunner(0xc0004c56c0, 0xc8f5f8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 127 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c5ba0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestDeleteAfterRead(0xc0004c5ba0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:968 +0x34
testing.tRunner(0xc0004c5ba0, 0xc8f5d8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 125 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c5860)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestFileBatchingRespectsStartAtEnd(0xc0004c5860)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:834 +0x34
testing.tRunner(0xc0004c5860, 0xc8f600)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 128 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0004c5d40)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestMaxBatching(0xc0004c5d40)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1042 +0x34
testing.tRunner(0xc0004c5d40, 0xc8f638)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 138 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0000c81a0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestStalePartialFingerprintDiscarded(0xc0000c81a0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1278 +0x34
testing.tRunner(0xc0000c81a0, 0xc8f6e8)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 1013 [chan send, 4 minutes]:
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest.NewSink.func1({0xc000262b10?, 0xc0004a84e2?}, {0xc0000ed000, 0x14, 0x0?}, 0xc000262a80)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:63 +0xfc
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader.(*Reader).ReadToEnd(0xc0003cb780, {0xdade48, 0xc000498280})
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader/reader.go:89 +0x520
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume.func1(0xc0003cb780)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:152 +0x8c
created by github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume in goroutine 48
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:[150](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694389961/job/23843241727?pr=32175#step:8:151) +0x1fc

goroutine 139 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0000c8340)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestWindowsFilesClosedImmediately(0xc0000c8340)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1317 +0x34
testing.tRunner(0xc0000c8340, 0xc8f720)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 140 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0000c84e0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestDelayedDisambiguation(0xc0000c84e0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1336 +0x34
testing.tRunner(0xc0000c84e0, 0xc8f5d0)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 141 [chan receive, 4 minutes]:
testing.(*T).Parallel(0xc0000c8680)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1483 +0x364
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestNoLostPartial(0xc0000c8680)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1377 +0x34
testing.tRunner(0xc0000c8680, 0xc8f688)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1689 +0x184
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/testing/testing.go:1742 +0x5e8

goroutine 48 [semacquire, 4 minutes]:
sync.runtime_Semacquire(0xc0004ca428?)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/runtime/sema.go:62 +0x2c
sync.(*WaitGroup).Wait(0xc0004ca420)
	/home/runner/actions-runner/_work/_tool/go/1.22.2/arm64/src/sync/waitgroup.go:116 +0x78
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume(0xc0000d20c0, {0xdade48, 0xc000498280}, {0xc0003cb300, 0x6, 0x8})
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:155 +0x384
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).poll(0xc0000d20c0, {0xdade48, 0xc000498280})
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:127 +0x380
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).startPoller.func1()
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:97 +0x124
created by github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).startPoller in goroutine 47
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:85 +0xe4

goroutine 1015 [chan send, 4 minutes]:
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest.NewSink.func1({0xc000262c90?, 0xc0004f6000?}, {0xc0000ef000, 0x13, 0x0?}, 0xc000262c00)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:63 +0xfc
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader.(*Reader).ReadToEnd(0xc0003cba80, {0xdade48, 0xc000498280})
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader/reader.go:89 +0x520
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume.func1(0xc0003cba80)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:[152](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694389961/job/23843241727?pr=32175#step:8:153) +0x8c
created by github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume in goroutine 48
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:150 +0x1fc

goroutine 1011 [chan send, 4 minutes]:
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest.NewSink.func1({0xc000262990?, 0xc0004ec000?}, {0xc0000e7000, 0x13, 0x3c?}, 0xc00028d5c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:63 +0xfc
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader.(*Reader).ReadToEnd(0xc0003cb480, {0xdade48, 0xc000498280})
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader/reader.go:89 +0x520
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume.func1(0xc0003cb480)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:152 +0x8c
created by github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume in goroutine 48
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file.go:150 +0x1fc
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer	300.055s

=== FAIL: fileconsumer TestMultiFileSort (unknown)

=== FAIL: fileconsumer TestFileBatching (unknown)

=== FAIL: fileconsumer TestFileBatchingRespectsStartAtEnd (unknown)

=== FAIL: fileconsumer TestWindowsFilesClosedImmediately (unknown)

=== FAIL: fileconsumer TestReadExistingAndNewLogs (unknown)

=== FAIL: fileconsumer TestNoNewline (unknown)

=== FAIL: fileconsumer TestLeadingEmpty (unknown)

=== FAIL: fileconsumer TestSplitWrite (unknown)

=== FAIL: fileconsumer TestDeleteAfterRead (unknown)

=== FAIL: fileconsumer TestMaxBatching (unknown)

=== FAIL: fileconsumer TestReadNewLogs (unknown)

=== FAIL: fileconsumer TestStartAtEndNewFile (unknown)

=== FAIL: fileconsumer TestMultiFileSortTimestamp (unknown)

=== FAIL: fileconsumer TestMultiFileParallel_PreloadedFiles (unknown)

=== FAIL: fileconsumer TestBuild (unknown)

=== FAIL: fileconsumer TestReadExistingLogs (unknown)

=== FAIL: fileconsumer TestReadUsingNopEncoding (unknown)

=== FAIL: fileconsumer TestMultipleEmpty (unknown)

=== FAIL: fileconsumer TestNoLostPartial (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc (unknown)

=== FAIL: fileconsumer TestMultiFileSimple (unknown)

=== FAIL: fileconsumer TestEncodings (unknown)

=== FAIL: fileconsumer TestStalePartialFingerprintDiscarded (unknown)

=== FAIL: fileconsumer TestNopEncodingDifferentLogSizes (unknown)

=== FAIL: fileconsumer TestStartAtEnd (unknown)

=== FAIL: fileconsumer TestMultiFileParallel_LiveFiles (unknown)

=== FAIL: fileconsumer TestManyLogsDelivered (unknown)

=== FAIL: fileconsumer TestDefaultBehaviors (unknown)

=== FAIL: fileconsumer TestIgnoreEmptyFiles (unknown)

=== FAIL: fileconsumer TestDecodeBufferIsResized (unknown)

=== FAIL: fileconsumer TestCopyTruncate (unknown)
    sink.go:114: 
        	Error Trace:	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:114
        	            				/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/rotation_test.go:83
        	Error:      	timeout: expected: 2700, actual: 863
        	Test:       	TestCopyTruncate

=== FAIL: fileconsumer TestEmptyLine (unknown)

=== FAIL: fileconsumer TestDelayedDisambiguation (unknown)

DONE 3219 tests, 34 failures in 309.528s
make[1]: Leaving directory '/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

--- a/Makefile.Common
+++ b/Makefile.Common
@@ -12,7 +12,7 @@ ifeq ($(UNIX_SHELL_ON_WINDOWS),true)
 	# The "sed" transformation below is needed on Windows, since commands like `go list -f '{{ .Dir }}'`
 	# return Windows paths and such paths are incompatible with other *nix tools, like `find`,
 	# used by the Makefile shell.
-	# The backslash needs to be doubled so its passed correctly to the shell. 
+	# The backslash needs to be doubled so its passed correctly to the shell.
 	NORMALIZE_DIRS = sed -e 's/^/\\//' -e 's/://' -e 's/\\\\/\\//g' | sort
 else
 	NORMALIZE_DIRS = sort
@@ -25,7 +25,7 @@ SRC_PARENT_DIR :=  $(shell dirname $(SRC_ROOT))
 
 # build tags required by any component should be defined as an independent variables and later added to GO_BUILD_TAGS below
 GO_BUILD_TAGS=""
-GOTEST_TIMEOUT?= 300s
+GOTEST_TIMEOUT?= 600s
 GOTEST_OPT?= -race -timeout $(GOTEST_TIMEOUT) -parallel 4 --tags=$(GO_BUILD_TAGS)
 GOTEST_INTEGRATION_OPT?= -race -timeout 360s -parallel 4
 GOTEST_OPT_WITH_COVERAGE = $(GOTEST_OPT) -coverprofile=coverage.txt -covermode=atomic
@@ -154,7 +154,7 @@ do-integration-tests-with-cover: $(GOTESTSUM)
 
 .PHONY: benchmark
 benchmark: $(GOTESTSUM)
-	$(GOTESTSUM) $(GOTESTSUM_OPT) --packages="$(ALL_PKGS)" -- -bench=. -run=notests --tags=$(GO_BUILD_TAGS) 
+	$(GOTESTSUM) $(GOTESTSUM_OPT) --packages="$(ALL_PKGS)" -- -bench=. -run=notests --tags=$(GO_BUILD_TAGS)
 
 .PHONY: addlicense
 addlicense: $(ADDLICENSE)
@@ -167,7 +167,7 @@ addlicense: $(ADDLICENSE)
 		exit 1; \
 	else \
 		echo "Add License finished successfully"; \
-	fi	
+	fi
 
 .PHONY: checklicense
 checklicense: $(ADDLICENSE)

